core/primitives/src/transaction.rs

```diff
@@ -366,11 +366,12 @@ pub struct ExecutionOutcome {
 
 #[derive(BorshSerialize, BorshDeserialize, PartialEq, Clone, Eq, Debug)]
 pub enum ExecutionMetadata {
-    // V1: Empty Metadata
+    /// V1: Empty Metadata
     V1,
-
-    // V2: With ProfileData
-    V2(ProfileData),
+    /// V2: With ProfileData by legacy `Cost` enum
+    V2(ProfileDataV1),
+    // V3: With ProfileData by gas parameters
+    V3(ProfileDataV2),
```

Comment:
Yes, I changed according to your suggestion. This means there is no `ProfileDataV1`, which makes sense since there is no profile data for `ExecutionMetadata::V1`.